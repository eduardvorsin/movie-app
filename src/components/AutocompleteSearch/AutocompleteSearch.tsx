'use client';

import Search, { Props as SearchProps } from '@/components/Search/Search';
import Link from "@/components/Link/Link";
import Spinner from '@/components/Spinner/Spinner';
import Title from '@/components/Title/Title';
import { ChangeEventHandler, FocusEventHandler, useState } from 'react';
import { MediaTypes } from '@/types/shared';

const iconAssets = {
	person: '/assets/icons/user.svg#user',
	tv: '/assets/icons/series.svg#series',
	movie: '/assets/icons/animation.svg#animation',
} as const;

export type AutocompleteOption = {
	href: string,
	label: string,
	iconType?: MediaTypes
}

export type Props = {
	isLoading?: boolean,
	onOptionLinkClick?: () => void,
	options: AutocompleteOption[],
	initialOptions?: AutocompleteOption[],
	dictionary: {
		emptyStateTitle: string,
		emptyStateText: string,
		search: Record<'button' | 'clearButton', string>,
	},
} & Omit<SearchProps, 'dictionary' | 'clearButton'>;

export default function AutocompleteSearch({
	id,
	className,
	testId,
	isDisabled,
	isReadOnly,
	isInvalid,
	isLoading,
	placeholder,
	options,
	initialOptions = [],
	label,
	labelHidden,
	error,
	name,
	value,
	onChange,
	onFocus,
	onBlur,
	onSubmit,
	onClear,
	onOptionLinkClick,
	dictionary,
}: Props) {
	const [isDropdownVisible, setIsDropdownVisible] = useState<boolean>(false);
	const currentOptions = value.length === 0 ? initialOptions : options;

	const blurHandler: FocusEventHandler<HTMLInputElement> = (e) => {
		setTimeout(() => setIsDropdownVisible(false), 100);
		if (onBlur) onBlur(e);
	};

	const focusHandler: FocusEventHandler<HTMLInputElement> = (e) => {
		if (initialOptions.length > 0 || value !== '') {
			setIsDropdownVisible(true);
		}

		if (onFocus) onFocus(e);
	};

	const changeHandler: ChangeEventHandler<HTMLInputElement> = (e) => {
		if (!isDropdownVisible && e.target.value !== '') {
			setIsDropdownVisible(true);
		}
		onChange(e);
	};

	const autocompleteClasses = [
		'mt-2 rounded-[0.1875rem] border-2 shadow-sm absolute w-full bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 transition-colors duration-150',
		isLoading ? 'min-h-[12.5rem]' : '',
	].join(' ');

	return (
		<div
			className={`relative ${className}`}
			data-testid={testId}
		>
			<Search
				id={id}
				name={name}
				isDisabled={isDisabled}
				isReadOnly={isReadOnly}
				isInvalid={isInvalid}
				placeholder={placeholder}
				label={label}
				labelHidden={labelHidden}
				value={value}
				onChange={changeHandler}
				onFocus={focusHandler}
				onBlur={blurHandler}
				onSubmit={onSubmit}
				onClear={onClear}
				clearButton={true}
				error={error}
				dictionary={dictionary.search}
			/>

			{isDropdownVisible && (
				<div
					className={autocompleteClasses}
				>
					{!isLoading && currentOptions.length === 0 && (
						<div
							className='p-[0.625rem] flex items-center justify-center flex-col text-neutral-1000 dark:text-dark-neutral-800 transition-colors duration-150'
						>
							<svg
								className='w-[5rem] h-[5rem] mb-5 fill-current'
								viewBox='0 0 32 32'
							>
								<use href={'/assets/icons/broken-loupe.svg#broken-loupe'}></use>
							</svg>

							<Title
								className='mb-3 break-all text-neutral-1000 dark:text-dark-neutral-800'
								level={5}
								as='h4'
								weight={400}
							>
								{dictionary.emptyStateTitle}
							</Title>

							<p className='text-neutral-900 dark:text-dark-neutral-700 transition-colors duration-150'>
								{dictionary.emptyStateText}
							</p>
						</div>
					)}

					{!isLoading && currentOptions.length !== 0 && (
						<ul>
							{currentOptions.map(({ label, iconType, href }) => (
								<li
									key={label}
									className='flex items-center px-1 border-b-2 last:border-none border-neutral-300 dark:border-dark-neutral-350 transition-colors duration-150'
								>
									{iconType && (
										<svg className='w-4 h-4 mr-2' viewBox='0 0 32 32'>
											<use href={iconAssets[iconType]}></use>
										</svg>
									)}

									<Link
										className='w-full h-full block px-[0.625rem] py-[0.375rem] [&]:bg-neutral-0 hover:[&]:bg-neutral-100 active:[&]:bg-neutral-200 dark:[&]:bg-dark-neutral-200 dark:hover:[&]:bg-dark-neutral-250 dark:active:[&]:bg-dark-neutral-300'
										href={href}
										onClick={onOptionLinkClick}
									>
										{label}
									</Link>
								</li>
							))}
						</ul>
					)}

					{isLoading && (
						<Spinner
							className='absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2'
							size='xlarge'
							testId={process.env.NODE_ENV === 'test' ? 'test-spinner' : undefined}
						/>
					)}
				</div>
			)}
		</div>
	);
}