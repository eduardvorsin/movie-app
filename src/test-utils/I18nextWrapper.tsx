import React, { ReactNode } from 'react';
import i18next from '@/i18n/client';
import { I18nextProvider } from 'react-i18next';

type Props = {
	children: ReactNode,
}

const I18nextWrapper = ({
	children,
}: Props) => {
	return (
		<I18nextProvider i18n={i18next}>
			{children}
		</I18nextProvider>
	);
};

export default I18nextWrapper;