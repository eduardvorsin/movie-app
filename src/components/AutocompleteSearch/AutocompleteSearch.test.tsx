import { render, screen } from "@testing-library/react";
import AutocompleteSearch from "./AutocompleteSearch";
import i18next from "@/i18n/client";
import I18nextWrapper from "@/test-utils/I18nextWrapper";
import userEvent from "@testing-library/user-event";

const options = [
	{
		href: '/',
		label: 'first option',
	}, {
		href: '/',
		label: 'second option',
	}, {
		href: '/',
		label: 'third option',
	}
];

describe('AutocompleteSearch tests', () => {
	it('is rendered correctly', async () => {
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn();
		const dictionary = {
			emptyStateTitle: 'We can\'t find anything for your request.',
			emptyStateText: 'Try entering a different value',
			search: {
				button: 'To find',
				clearButton: 'To clear',
			},
		};
		render(
			<AutocompleteSearch
				id='test-autocomplete-search'
				label='autocomplete-search'
				name='autocomplete-search'
				onChange={changeMockFn}
				onSubmit={submitMockFn}
				value=''
				options={options}
				initialOptions={[]}
				testId='test-autocomplete-search'
				dictionary={dictionary}
			/>
		);

		await user.click(screen.getByRole<HTMLInputElement>('textbox'));

		expect(screen.getByTestId<HTMLDivElement>('test-autocomplete-search')).toBeInTheDocument();
	});

	it('if isLoading is true, the spinner should be displayed', async () => {
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn();
		const dictionary = {
			emptyStateTitle: 'We can\'t find anything for your request.',
			emptyStateText: 'Try entering a different value',
			search: {
				button: 'To find',
				clearButton: 'To clear',
			},
		};
		render(
			<AutocompleteSearch
				id='test-autocomplete-search'
				label='autocomplete search'
				name='autocomplete-search'
				onChange={changeMockFn}
				onSubmit={submitMockFn}
				value=''
				options={[]}
				initialOptions={[]}
				testId='test-autocomplete-search'
				dictionary={dictionary}
				isLoading
			/>
		);

		await user.type(screen.getByRole<HTMLInputElement>('textbox'), ' ');

		expect(screen.getByTestId<HTMLSpanElement>('test-spinner')).toBeInTheDocument();
	});

	it('If nothing is found at the user\'s request, then the corresponding message should be displayed', async () => {
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn();
		const dictionary = {
			emptyStateTitle: 'We can\'t find anything for your request.',
			emptyStateText: 'Try entering a different value',
			search: {
				button: 'To find',
				clearButton: 'To clear',
			},
		};
		render(
			<AutocompleteSearch
				id='test-autocomplete-search'
				label='autocomplete-search'
				name='autocomplete-search'
				onChange={changeMockFn}
				onSubmit={submitMockFn}
				value=''
				options={[]}
				initialOptions={[]}
				testId='test-autocomplete-search'
				dictionary={dictionary}
			/>
		);

		await user.type(screen.getByRole<HTMLInputElement>('textbox'), ' ');

		expect(screen.getByText<HTMLParagraphElement>(dictionary.emptyStateText)).toBeInTheDocument();
		expect(screen.getByRole<HTMLHeadingElement>('heading', { name: dictionary.emptyStateTitle })).toBeInTheDocument();
	});

	it('is a basic snapshot', async () => {
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn();
		const dictionary = {
			emptyStateTitle: 'We can\'t find anything for your request.',
			emptyStateText: 'Try entering a different value',
			search: {
				button: 'To find',
				clearButton: 'To clear',
			},
		};
		render(
			<AutocompleteSearch
				id='test-autocomplete-search'
				label='autocomplete-search'
				name='autocomplete-search'
				onChange={changeMockFn}
				onSubmit={submitMockFn}
				value=''
				options={options}
				initialOptions={[]}
				testId='test-autocomplete-search'
				dictionary={dictionary}
			/>
		);

		await user.click(screen.getByRole<HTMLInputElement>('textbox'));

		expect(screen.getByTestId<HTMLDivElement>('test-autocomplete-search')).toMatchSnapshot();
	});

	it('is a snapshot with isLoading is true', async () => {
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn();
		const dictionary = {
			emptyStateTitle: 'We can\'t find anything for your request.',
			emptyStateText: 'Try entering a different value',
			search: {
				button: 'To find',
				clearButton: 'To clear',
			},
		};
		render(
			<AutocompleteSearch
				id='test-autocomplete-search'
				label='autocomplete-search'
				name='autocomplete-search'
				onChange={changeMockFn}
				onSubmit={submitMockFn}
				value=''
				options={[]}
				initialOptions={[]}
				testId='test-autocomplete-search'
				dictionary={dictionary}
				isLoading
			/>
		);

		await user.click(screen.getByRole<HTMLInputElement>('textbox'));

		expect(screen.getByTestId<HTMLDivElement>('test-autocomplete-search')).toMatchSnapshot();
	});

	it('is a snapshot when initialOptions and options are empty', async () => {
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn();
		const dictionary = {
			emptyStateTitle: 'We can\'t find anything for your request.',
			emptyStateText: 'Try entering a different value',
			search: {
				button: 'To find',
				clearButton: 'To clear',
			},
		};
		render(
			<AutocompleteSearch
				id='test-autocomplete-search'
				label='autocomplete-search'
				name='autocomplete-search'
				onChange={changeMockFn}
				onSubmit={submitMockFn}
				value=''
				options={[]}
				initialOptions={[]}
				testId='test-autocomplete-search'
				dictionary={dictionary}
			/>
		);

		await user.click(screen.getByRole<HTMLInputElement>('textbox'));

		expect(screen.getByTestId<HTMLDivElement>('test-autocomplete-search')).toMatchSnapshot();
	});
});

describe('AutocompleteSearch integration tests', () => {
	it('localization into English works correctly', async () => {
		await i18next.changeLanguage('en');
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn();
		const translation = new RegExp(i18next.t('autocompleteSearch.emptyStateText'));

		const dictionary = {
			emptyStateTitle: 'We can\'t find anything for your request.',
			emptyStateText: 'Try entering a different value',
			search: {
				button: 'To find',
				clearButton: 'To clear',
			},
		};
		render(
			<AutocompleteSearch
				id='test-autocomplete-search'
				label='autocomplete-search'
				name='autocomplete-search'
				onChange={changeMockFn}
				onSubmit={submitMockFn}
				value=''
				options={[]}
				initialOptions={[]}
				testId='test-autocomplete-search'
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		await user.type(screen.getByRole<HTMLInputElement>('textbox'), ' ');

		expect(screen.getByText<HTMLParagraphElement>(translation)).toBeInTheDocument();
	});

	it('localization into Russian works correctly', async () => {
		await i18next.changeLanguage('ru');
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn();
		const translation = new RegExp(i18next.t('autocompleteSearch.emptyStateText'));

		const dictionary = {
			emptyStateTitle: 'Мы ничего не смогли найти по вашему запросу.',
			emptyStateText: 'Попробуйте ввести другое значение.',
			search: {
				button: 'Найти',
				clearButton: 'Очистить',
			},
		};
		render(
			<AutocompleteSearch
				id='test-autocomplete-search'
				label='autocomplete-search'
				name='autocomplete-search'
				onChange={changeMockFn}
				onSubmit={submitMockFn}
				value=''
				options={[]}
				initialOptions={[]}
				testId='test-autocomplete-search'
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		await user.type(screen.getByRole<HTMLInputElement>('textbox'), ' ');

		expect(screen.getByText<HTMLParagraphElement>(translation)).toBeInTheDocument();
	});

	it('is a snapshot with English localization', async () => {
		await i18next.changeLanguage('en');
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn();

		const dictionary = {
			emptyStateTitle: 'We can\'t find anything for your request.',
			emptyStateText: 'Try entering a different value',
			search: {
				button: 'To find',
				clearButton: 'To clear',
			},
		};
		render(
			<AutocompleteSearch
				id='test-autocomplete-search'
				label='autocomplete-search'
				name='autocomplete-search'
				onChange={changeMockFn}
				onSubmit={submitMockFn}
				value=''
				options={[]}
				initialOptions={[]}
				testId='test-autocomplete-search'
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		await user.type(screen.getByRole<HTMLInputElement>('textbox'), ' ');

		expect(screen.getByTestId<HTMLDivElement>('test-autocomplete-search')).toMatchSnapshot();
	});

	it('is a snapshot with Russian localization', async () => {
		await i18next.changeLanguage('ru');
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn();

		const dictionary = {
			emptyStateTitle: 'Мы ничего не смогли найти по вашему запросу.',
			emptyStateText: 'Попробуйте ввести другое значение.',
			search: {
				button: 'Найти',
				clearButton: 'Очистить',
			},
		};
		render(
			<AutocompleteSearch
				id='test-autocomplete-search'
				label='autocomplete-search'
				name='autocomplete-search'
				onChange={changeMockFn}
				onSubmit={submitMockFn}
				value=''
				options={[]}
				initialOptions={[]}
				testId='test-autocomplete-search'
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		await user.type(screen.getByRole<HTMLInputElement>('textbox'), ' ');

		expect(screen.getByTestId<HTMLDivElement>('test-autocomplete-search')).toMatchSnapshot();
	});
});