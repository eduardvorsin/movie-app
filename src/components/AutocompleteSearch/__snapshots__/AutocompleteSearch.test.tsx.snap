// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteSearch integration tests is a snapshot with English localization 1`] = `
<div
  class="relative undefined"
  data-testid="test-autocomplete-search"
>
  <div>
    <form
      autocomplete="off"
      role="search"
    >
      <label>
        <span
          class="block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 transition-colors duration-150 "
        >
          autocomplete-search
        </span>
        <div
          class="relative"
        >
          <input
            class="outline-none w-full h-[2.25rem] py-2 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150   border-neutral-300 dark:border-dark-neutral-350 px-8"
            id="test-autocomplete-search"
            inputmode="search"
            name="autocomplete-search"
            type="text"
            value=""
          />
          <button
            class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   [&]:text-neutral-400 enabled:hover:[&]:text-neutral-300 enabled:active:[&]:text-neutral-200 dark:[&]:text-dark-neutral-600 dark:enabled:hover:text-dark-neutral-700 dark:enabled:active:text-dark-neutral-800 absolute left-[0.375rem] top-1/2 -translate-y-1/2"
            type="button"
          >
            <span
              class="inline-flex items-center"
            >
              <svg
                class="w-6 h-6 fill-current"
                viewBox="0 0 32 32"
              >
                <use
                  href="/assets/icons/cancel.svg#cancel"
                />
              </svg>
              To clear
            </span>
          </button>
          <button
            class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600   absolute top-1/2 right-[0.375rem] -translate-y-1/2"
          >
            <span
              class="inline-flex items-center"
            >
              <svg
                class="w-6 h-6 p-0.5 fill-current"
                viewBox="0 0 32 32"
              >
                <use
                  href="/assets/icons/loupe.svg#loupe"
                />
              </svg>
              To find
            </span>
          </button>
        </div>
      </label>
    </form>
  </div>
  <div
    class="mt-2 rounded-[0.1875rem] border-2 shadow-sm absolute w-full bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 transition-colors duration-150 "
  >
    <div
      class="p-[0.625rem] flex items-center justify-center flex-col text-neutral-1000 dark:text-dark-neutral-800 transition-colors duration-150"
    >
      <svg
        class="w-[5rem] h-[5rem] mb-5 fill-current"
        viewBox="0 0 32 32"
      >
        <use
          href="/assets/icons/broken-loupe.svg#broken-loupe"
        />
      </svg>
      <h4
        class="leading-[1.25] -tracking-[0.01em] transition-colors duration-150 text-300 font-regular mb-3 break-all text-neutral-1000 dark:text-dark-neutral-800"
      >
        We can't find anything for your request.
      </h4>
      <p
        class="text-neutral-900 dark:text-dark-neutral-700 transition-colors duration-150"
      >
        Try entering a different value
      </p>
    </div>
  </div>
</div>
`;

exports[`AutocompleteSearch integration tests is a snapshot with Russian localization 1`] = `
<div
  class="relative undefined"
  data-testid="test-autocomplete-search"
>
  <div>
    <form
      autocomplete="off"
      role="search"
    >
      <label>
        <span
          class="block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 transition-colors duration-150 "
        >
          autocomplete-search
        </span>
        <div
          class="relative"
        >
          <input
            class="outline-none w-full h-[2.25rem] py-2 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150   border-neutral-300 dark:border-dark-neutral-350 px-8"
            id="test-autocomplete-search"
            inputmode="search"
            name="autocomplete-search"
            type="text"
            value=""
          />
          <button
            class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   [&]:text-neutral-400 enabled:hover:[&]:text-neutral-300 enabled:active:[&]:text-neutral-200 dark:[&]:text-dark-neutral-600 dark:enabled:hover:text-dark-neutral-700 dark:enabled:active:text-dark-neutral-800 absolute left-[0.375rem] top-1/2 -translate-y-1/2"
            type="button"
          >
            <span
              class="inline-flex items-center"
            >
              <svg
                class="w-6 h-6 fill-current"
                viewBox="0 0 32 32"
              >
                <use
                  href="/assets/icons/cancel.svg#cancel"
                />
              </svg>
              Очистить
            </span>
          </button>
          <button
            class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600   absolute top-1/2 right-[0.375rem] -translate-y-1/2"
          >
            <span
              class="inline-flex items-center"
            >
              <svg
                class="w-6 h-6 p-0.5 fill-current"
                viewBox="0 0 32 32"
              >
                <use
                  href="/assets/icons/loupe.svg#loupe"
                />
              </svg>
              Найти
            </span>
          </button>
        </div>
      </label>
    </form>
  </div>
  <div
    class="mt-2 rounded-[0.1875rem] border-2 shadow-sm absolute w-full bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 transition-colors duration-150 "
  >
    <div
      class="p-[0.625rem] flex items-center justify-center flex-col text-neutral-1000 dark:text-dark-neutral-800 transition-colors duration-150"
    >
      <svg
        class="w-[5rem] h-[5rem] mb-5 fill-current"
        viewBox="0 0 32 32"
      >
        <use
          href="/assets/icons/broken-loupe.svg#broken-loupe"
        />
      </svg>
      <h4
        class="leading-[1.25] -tracking-[0.01em] transition-colors duration-150 text-300 font-regular mb-3 break-all text-neutral-1000 dark:text-dark-neutral-800"
      >
        Мы ничего не смогли найти по вашему запросу.
      </h4>
      <p
        class="text-neutral-900 dark:text-dark-neutral-700 transition-colors duration-150"
      >
        Попробуйте ввести другое значение.
      </p>
    </div>
  </div>
</div>
`;

exports[`AutocompleteSearch tests is a basic snapshot 1`] = `
<div
  class="relative undefined"
  data-testid="test-autocomplete-search"
>
  <div>
    <form
      autocomplete="off"
      role="search"
    >
      <label>
        <span
          class="block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 transition-colors duration-150 "
        >
          autocomplete-search
        </span>
        <div
          class="relative"
        >
          <input
            class="outline-none w-full h-[2.25rem] py-2 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150   border-neutral-300 dark:border-dark-neutral-350 px-8"
            id="test-autocomplete-search"
            inputmode="search"
            name="autocomplete-search"
            type="text"
            value=""
          />
          <button
            class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   [&]:text-neutral-400 enabled:hover:[&]:text-neutral-300 enabled:active:[&]:text-neutral-200 dark:[&]:text-dark-neutral-600 dark:enabled:hover:text-dark-neutral-700 dark:enabled:active:text-dark-neutral-800 absolute left-[0.375rem] top-1/2 -translate-y-1/2"
            type="button"
          >
            <span
              class="inline-flex items-center"
            >
              <svg
                class="w-6 h-6 fill-current"
                viewBox="0 0 32 32"
              >
                <use
                  href="/assets/icons/cancel.svg#cancel"
                />
              </svg>
              To clear
            </span>
          </button>
          <button
            class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600   absolute top-1/2 right-[0.375rem] -translate-y-1/2"
          >
            <span
              class="inline-flex items-center"
            >
              <svg
                class="w-6 h-6 p-0.5 fill-current"
                viewBox="0 0 32 32"
              >
                <use
                  href="/assets/icons/loupe.svg#loupe"
                />
              </svg>
              To find
            </span>
          </button>
        </div>
      </label>
    </form>
  </div>
</div>
`;

exports[`AutocompleteSearch tests is a snapshot when initialOptions and options are empty 1`] = `
<div
  class="relative undefined"
  data-testid="test-autocomplete-search"
>
  <div>
    <form
      autocomplete="off"
      role="search"
    >
      <label>
        <span
          class="block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 transition-colors duration-150 "
        >
          autocomplete-search
        </span>
        <div
          class="relative"
        >
          <input
            class="outline-none w-full h-[2.25rem] py-2 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150   border-neutral-300 dark:border-dark-neutral-350 px-8"
            id="test-autocomplete-search"
            inputmode="search"
            name="autocomplete-search"
            type="text"
            value=""
          />
          <button
            class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   [&]:text-neutral-400 enabled:hover:[&]:text-neutral-300 enabled:active:[&]:text-neutral-200 dark:[&]:text-dark-neutral-600 dark:enabled:hover:text-dark-neutral-700 dark:enabled:active:text-dark-neutral-800 absolute left-[0.375rem] top-1/2 -translate-y-1/2"
            type="button"
          >
            <span
              class="inline-flex items-center"
            >
              <svg
                class="w-6 h-6 fill-current"
                viewBox="0 0 32 32"
              >
                <use
                  href="/assets/icons/cancel.svg#cancel"
                />
              </svg>
              To clear
            </span>
          </button>
          <button
            class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600   absolute top-1/2 right-[0.375rem] -translate-y-1/2"
          >
            <span
              class="inline-flex items-center"
            >
              <svg
                class="w-6 h-6 p-0.5 fill-current"
                viewBox="0 0 32 32"
              >
                <use
                  href="/assets/icons/loupe.svg#loupe"
                />
              </svg>
              To find
            </span>
          </button>
        </div>
      </label>
    </form>
  </div>
</div>
`;

exports[`AutocompleteSearch tests is a snapshot with isLoading is true 1`] = `
<div
  class="relative undefined"
  data-testid="test-autocomplete-search"
>
  <div>
    <form
      autocomplete="off"
      role="search"
    >
      <label>
        <span
          class="block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 transition-colors duration-150 "
        >
          autocomplete-search
        </span>
        <div
          class="relative"
        >
          <input
            class="outline-none w-full h-[2.25rem] py-2 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150   border-neutral-300 dark:border-dark-neutral-350 px-8"
            id="test-autocomplete-search"
            inputmode="search"
            name="autocomplete-search"
            type="text"
            value=""
          />
          <button
            class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   [&]:text-neutral-400 enabled:hover:[&]:text-neutral-300 enabled:active:[&]:text-neutral-200 dark:[&]:text-dark-neutral-600 dark:enabled:hover:text-dark-neutral-700 dark:enabled:active:text-dark-neutral-800 absolute left-[0.375rem] top-1/2 -translate-y-1/2"
            type="button"
          >
            <span
              class="inline-flex items-center"
            >
              <svg
                class="w-6 h-6 fill-current"
                viewBox="0 0 32 32"
              >
                <use
                  href="/assets/icons/cancel.svg#cancel"
                />
              </svg>
              To clear
            </span>
          </button>
          <button
            class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600   absolute top-1/2 right-[0.375rem] -translate-y-1/2"
          >
            <span
              class="inline-flex items-center"
            >
              <svg
                class="w-6 h-6 p-0.5 fill-current"
                viewBox="0 0 32 32"
              >
                <use
                  href="/assets/icons/loupe.svg#loupe"
                />
              </svg>
              To find
            </span>
          </button>
        </div>
      </label>
    </form>
  </div>
</div>
`;
